sys/kern/src/arch/rv64/task.rs

```diff
@@ -20,31 +43,27 @@ use unwrap_lite::UnwrapLite;
 pub unsafe fn set_current_task(task: &task::Task) {
     // Safety: should be ok if the contract above is met
     // TODO: make me an atomic
-    unsafe {
-        let task = task as *const task::Task as usize;
-        register::mscratch::write(task);
-    }
+    let task = task as *const task::Task as usize;
+
+    xscratch::write(task);
 }
 
 pub unsafe fn get_current_task() -> &'static task::Task {
-    unsafe {
-        let task = register::mscratch::read();
-        uassert!(task != 0);
-        &*(task as *const task::Task)
-    }
+    let task = xscratch::read();
+    uassert!(task != 0);
+    unsafe { &*(task as *const task::Task) }
```

Comment:
nit: This is a redundant unsafe

Reply:
Unfortunately, the compiler disagrees:

`error: dereference of raw pointer is unsafe and requires unsafe block (error E0133)
  --> sys/kern/src/arch/rv64/task.rs:54:5
   |
54 |     &*(task as *const task::Task)
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ dereference of raw pointer
   |`